pkg/scheduler/framework/plugins/interpodaffinity/filtering.go

```diff
@@ -134,6 +140,46 @@ func (m topologyToMatchedTermCount) updateWithAntiAffinityTerms(terms []framewor
 	}
 }
 
+// topologyMatchedTermCountList is a slice equivalent of topologyToMatchedTermCount map.
+type topologyMatchedTermCountList []topologyMatchedTermCount
+
+type topologyMatchedTermCount struct {
+	topologyPair topologyPair
+	count        int64
+}
+
+func (m *topologyMatchedTermCountList) append(node *v1.Node, tk string, value int64) {
+	if tv, ok := node.Labels[tk]; ok {
+		pair := topologyPair{key: tk, value: tv}
+		*m = append(*m, topologyMatchedTermCount{
+			topologyPair: pair,
+			count:        value,
+		})
+	}
+}
+
+// appends the specified value to the topologyMatchedTermCountList
+// for each affinity term if "targetPod" matches ALL terms.
+func (m *topologyMatchedTermCountList) appendWithAffinityTerms(
```

Comment:
I don't like the fact the methods are in fact duplicated, but don't have any reasonable aternative here.